### PR TITLE
Hint to remove parentheses that would result in a parse error

### PR DESCRIPTION
**Pull request by elland**
Addresses https://github.com/ndmitchell/hlint/issues/1501

### CODE REVIEW

File: src/Hint/Bracket.hs [hunk 1]
```diff
@@ -120,6 +120,9 @@ foo' :: ((A) => ()) -> ()
 bar' :: ((A a) => ()) -> ()
 data Dict c where Dict :: (c) => Dict c
 data Dict' c a where Dict' :: (c a) => Dict' c a
+
+issue1501: Redundant bracket hint resulted in a parse error
```
Comment (zliu41) [inline src/Hint/Bracket.hs:124]:
```suggestion
-- issue1501: Redundant bracket hint resulted in a parse error
```

File: src/Hint/Bracket.hs [hunk 2]
```diff
@@ -120,6 +120,9 @@ foo' :: ((A) => ()) -> ()
...
+\(Proxy @a) -> True
```
Comment (zliu41) [inline src/Hint/Bracket.hs:125]: You may need a binding, as in `x = \(Proxy @a) -> True` - I'm not sure.

Comment (zliu41) [reply src/Hint/Bracket.hs:125]: Actually this would cause HLint to suggest `x (Proxy @a) = True`. You could use `x = f $ \(Proxy @a) -> True`.

Comment (elland) [reply src/Hint/Bracket.hs:125]: Do you mean as an extra case or instead of the one already there? :)

Comment (zliu41) [reply src/Hint/Bracket.hs:125]: I meant instead of. This is not needed given the one below.

**Review by zliu41 — APPROVED** [review]
Thanks!

**Pull request merged**